src/odemis/driver/simcam.py

```diff
@@ -142,6 +143,48 @@ def __init__(self, name, role, image, dependencies=None, daemon=None, blur_facto
         # Convenience event for the user to connect and fire
         self.softwareTrigger = model.Event()
 
+        # show_message(None, 'starting and error wise thread', 'test error', timeout = 1.0)
+        self._errorCreationThread = threading.Thread(target = self._state_error_creation,
+                                                     args = (),
+                                                     name = "Creating and state error")
+        self._errorCreationThread.daemon = True
+        self._errorCreationThread.start()
+
+    def _state_error_creation(self):
+        '''
+        by creating the files 'simcam.HW.error' or 'simcam.HW.good' in the folder '/var/run/odemisd' and hardware
+        error can be produced in the Sim Cam or the error can be 'fixed'
+        After the error is produced the files should be deleted
+        :return:
+        '''
```

Comment:
For the main function of thread, I'd recommend to surround the entire function by a try/except, so that if an exception happens, you can log the exception (instead of it being sent to a black hole).

Reply:
do you mean to include the part:

`            if os.path.isfile(path + '/simcam.HW.error'):
                self.state._set_value(model.HwError("Camera disconnected. Now recovering..."), force_write = True)
                os.remove(path + '/simcam.HW.error')

            elif os.path.isfile(path + '/simcam.HW.good'):
                self.state._set_value('running', force_write = True)
                os.remove(path + '/simcam.HW.good')`
in a try expect block?

Reply:
I mean that the first line of the function should be `try:`, and the *entire* current code should be within that block.